src/main/java/org/jmeterplugins/repository/JARSourceHTTP.java

```diff
@@ -147,6 +178,79 @@ protected JSON getJSON(String uri) throws IOException {
         }
     }
 
+    private PluginsRepo getRepoCache(String uri) {
+        File file = generateCacheFile(uri);
+        if (!file.exists()) {
+            return null;
+        }
+
+        return SerializeUtils.deserialize(file);
+    }
+
+    private void cacheRepo(String repoJSON, HttpResponse response, String uri) {
+        // default cache expire is 1 hour since now
+        long maxAge = CACHE_MAX_AGE;
+        long date = System.currentTimeMillis();
+
+        Header[] allHeaders = response.getAllHeaders();
+        for (Header header : allHeaders) {
+            if ("date".equals(header.getName().toLowerCase())) {
+                date = parseDateHeader(header);
+            } else if ("cache-control".equals(header.getName().toLowerCase())) {
+                maxAge = parseCacheControlHeader(header);
+            }
+        }
+
+        long expirationTime = date + maxAge;
+        PluginsRepo repo = new PluginsRepo(repoJSON, expirationTime);
+        SerializeUtils.serialize(repo, generateCacheFile(uri));
+    }
+
+    private File generateCacheFile(String uri) {
+        return new File(cacheDir, generateFileName(uri));
+    }
+
+    private String generateFileName(String uri) {
```

Comment:
Why not just generate md5 from URI and don't mess with words?

Reply:
And also add `System.getProperty("user.name")` to that MD5, so files of different users won't clash